### ISSUE TITLE
Handle JSX Pragma

### ISSUE THREAD

**Issue by thebuilder** [simonhaenisch/prettier-plugin-organize-imports#12]
While generally import order doesn't matter, there are some cases where it does. One of those is when importing a custom [JSX pragma](https://www.gatsbyjs.com/blog/2019-08-02-what-is-jsx-pragma/). In those cases, the file contains a `@jsx` comment before importing the JSX pragma.

```ts
/** @jsx jsx */
import { jsx } from './jsx';
import { ElementType, forwardRef } from 'react';
import { ResponsiveValue, system } from 'styled-system';
```

Organizing the imports on this file, results in a new order that breaks the code:

```ts
/** @jsx jsx */
import { ElementType, forwardRef } from 'react';
import { ResponsiveValue, system } from 'styled-system';
import { jsx } from './jsx';
```

I'm not sure if this is something this library can even fix, or if it would be part of the TypeScript implementation?

**Comment by thebuilder** [comment]
Found a related issue:
https://github.com/microsoft/TypeScript/issues/28827

Might not be something you can fix.

**Comment by simonhaenisch** [comment]
This is related to #9. The problem is that the first comment line is considered a file header comment by `organizeImports` and therefore doesn't get moved along with the import statement. There's a lot of discussion going on about this but no concrete ideas for a solution yet (I'd prefer if the first comment was only considered a file header if it's followed by a blank line).

If you had the comment not as the first line, then it would move the pragma comment with the import statement, e. g.

```ts
import { ElementType, forwardRef } from 'react';
import { ResponsiveValue, system } from 'styled-system';
/** @jsx jsx */
import { jsx } from './jsx';
```

I don't know much about those pragma comments though. TBH I'm wondering: does the pragma comment actually need to be before the `jsx` import statement? I wouldn't think so... I would do

```ts
/** @jsx jsx */

import { ElementType, forwardRef } from 'react';
import { ResponsiveValue, system } from 'styled-system';
import { jsx } from './jsx';
```

And I'd assume that works? Anyway, you should also be able to set `jsxFactory` in tsconfig instead.

TL;DR it's not up to this plugin to fix this.
